index.js

```diff
@@ -1,22 +1,43 @@
+var path = require('path');
+
 module.exports = {
   name: 'ember-cli-6to5',
+
+  treeFor: function(name) {
+    if (name !== 'vendor') { return; }
+
+    // Use the transpiler's node_modules directory the vendor tree for this addon
+    return this.treeGenerator(path.join(__dirname, 'node_modules', 'broccoli-6to5-transpiler', 'node_modules'));
+  },
+
   included: function(app) {
     this._super.included.apply(this, arguments);
 
     var options = getOptions(app.options['6to5']);
 
-    var plugin = {
-      name: 'ember-cli-6to5',
-      ext: 'js',
-      toTree: function(tree) {
-        return require('broccoli-6to5-transpiler')(tree, options);
-      }
-    };
+    registerPlugin(app, options);
 
-    app.registry.add('js', plugin);
+    if (options.includePolyfill) {
+      delete options.includePolyfill;
+      includePolyfill(app);
+    }
   }
 };
 
+function registerPlugin(app, options) {
+  app.registry.add('js', {
+    name: 'ember-cli-6to5',
+    ext: 'js',
+    toTree: function(tree) {
+      return require('broccoli-6to5-transpiler')(tree, options);
+    }
+  });
+}
+
+function includePolyfill(app) {
+  app.import('vendor/6to5-core/browser-polyfill.js');
```

Comment:
Good point; I hadn't thought about the complications from deduping. Seems like this would also impact [ember-cli-ic-ajax's implementation](https://github.com/rwjblue/ember-cli-ic-ajax/blob/master/index.js#L9), which was mentioned in some thread as the canonical way to handle this kind of import. The likelihood of ic-ajax getting hoisted is probably a lot lower, though, given that there aren't many other packages out there that depend on it.